### PR TITLE
(ISA-264) State of knowledge layers tweaks

### PR DESCRIPTION
**Pull request by JoshGx2000**
[ISA-264](https://jira.its.utas.edu.au/browse/ISA-264)

Nothing major, but have got to remember to update the code for `:add-layer-below-layer` PR #120 goes through.

### CODE REVIEW

File: frontend/src/cljs/imas_seamap/utils.cljs [hunk 1]
```diff
@@ -237,3 +237,10 @@
     (doseq [[k v] params]
       (.setParameterValue u (name k) v))
     (str u)))
+
+(defn index-of [pred coll]
+  (map
+   first
+   (filter
+    #(pred (second %))
+    (map-indexed vector coll))))
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/utils.cljs:246]: Definitely needs a docstring!

File: frontend/src/cljs/imas_seamap/map/events.cljs [hunk 2]
```diff
@@ -537,6 +537,21 @@
...
+        index         (first (index-of #(= % target-layer) active-layers))
+        below         (subvec active-layers 0 index)
+        above         (subvec active-layers index)
+        db            (if-not ((set active-layers) layer)
+                        (assoc-in db [:map :active-layers] (vec (concat below [layer] above)))
+                        db)]
+    {:db       db
+     :put-hash (encode-state db)
+     :dispatch [:map/popup-closed]}))
+
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/map/events.cljs:554]: I suppose the main thing I wonder is: is it simpler/clearer to just make `:add-layer` take an optional arg, for example?  (or something along those lines)

**Review by markhepburn — CHANGES_REQUESTED** [review]
One small change request / one question for consideration.

File: frontend/src/cljs/imas_seamap/map/events.cljs (hunk 2 above)
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/map/events.cljs:554]: Yes, that's a good idea. Definitely worth it to not pollute the ns!

File: frontend/src/cljs/imas_seamap/utils.cljs (hunk 1 above)
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/utils.cljs]: Fixed, and using the thread-last macro for neater code!

File: frontend/src/cljs/imas_seamap/map/events.cljs (hunk 2 above)
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/map/events.cljs]: Done!

File: frontend/src/cljs/imas_seamap/utils.cljs [hunk 3]
```diff
@@ -237,3 +237,11 @@
...
       (.setParameterValue u (name k) v))
     (str u)))
+
+(defn index-of
+  "Returns a list of indexes that match the predicate within the collection."
+  [pred coll]
+  (->> coll
+       (map-indexed vector)
+       (filter #(pred (second %)))
+       (map first)))
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/utils.cljs:247]: Much clearer :)

**Pull request merged**